modules/openapi/src/internals/OpenApiToIModel.scala

```diff
@@ -362,16 +364,17 @@ private class OpenApiToIModel[F[_]: Parallel: TellShape: TellError](
   private def recordService(opDefIds: Vector[DefId]): F[Unit] = {
     val security =
       Option(openApi.getSecurity())
-        .map(_.asScala)
+        .map(_.asScala.toSet)
         .getOrElse(Set.empty)
         .flatMap(_.keySet().asScala)
     val auth = security.flatMap(securitySchemes.get).toVector
     val authHint = if (auth.nonEmpty) List(Hint.Auth(auth)) else Nil
     val schemes = securitySchemes.values.toVector
     val securityHint =
       if (schemes.nonEmpty) List(Hint.Security(schemes)) else Nil
-    val exts = GetExtensions.from(openApi)
-    val infoExts = GetExtensions.from(openApi.getInfo())
+    val exts = GetExtensions.from(openApi.asInstanceOf[HasExtensions])
```

Comment:
EDIT : yeah okay, I had forgotten it was a reflective call that we need because the openapi library is horrendous. 

I guess e could centralise the cast in a `def unsafeFrom(a: Any)`